tests/unit/test_platforms.py

```diff
@@ -414,3 +428,28 @@ def test_get_install_target_to_platforms_map(
                 _map[install_target] = sorted(_map[install_target],
                                               key=lambda k: k['name'])
         assert result == expected_map
+
+
+@pytest.mark.parametrize(
+    'platform, job, remote, expect',
+    [
+        (
+            # Default, no old settings.
+            {'ship': 'Enterprise'}, {}, {}, True
+        ),
+        (
+            {'captain': 'Kirk'},
+            {'captain': 'Picard'},
+            {},
+            False
+        ),
+        (
+            {'captain': 'Sisko'},
+            {},
+            {'captain': 'Janeway'},
+            False
+        )
```

Comment:
I know it's a simple piece of logic being tested, but for completeness, what about a couple more cases like:
```
(
    {'captain': 'Picard', 'ship': 'Enterprise'},
    {'captain': 'Picard'},
    {'ship': 'Enterprise'},
    True
),
(
    {'captain': 'Picard', 'ship': 'Enterprise'},
    {'captain': 'Picard'},
    {'ship': 'Defiant'},
    False
)
```
(if that's indeed what the expected return values should be, I haven't checked)

Reply:
Yes please. These both look like reasonable cases.